%3a117/list-queues-impl.scm

```diff
@@ -0,0 +1,237 @@
+;;;; Implementation of list-queue SRFI
+;;;;
+;;;; Copyright (C) John Cowan (2015, 2016). All Rights Reserved.
+;;;;
+;;;; Permission is hereby granted, free of charge, to any person obtaining a copy of
+;;;; this software and associated documentation files (the "Software"), to deal in
+;;;; the Software without restriction, including without limitation the rights to
+;;;; use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+;;;; of the Software, and to permit persons to whom the Software is furnished to do
+;;;; so, subject to the following conditions:
+;;;;
+;;;; The above copyright notice and this permission notice shall be included in all
+;;;; copies or substantial portions of the Software.
+;;;;
+;;;; THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+;;;; IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+;;;; FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+;;;; AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+;;;; LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+;;;; OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+;;;; SOFTWARE.
+
+;;;; Later modified from the R7RS SRFI-117 reference implementation for
+;;;; possible inclusion within the R6RS https://github.com/arcfide/chez-srfi.
+
+;;; R7RS shims.  Comment these out on an R7RS system.
+;;; [John Cowan] stole this code from Chibi Scheme, which is BSD-licensed.
```

Comment:
s/I/[John Cowan]/ as can be confirmed from https://github.com/scheme-requests-for-implementation/srfi-117/blob/master/list-queues/list-queues-impl.scm#L4.